### ISSUE TITLE
Examples for SC 2.1.4 too faint - difficult to discern

### ISSUE THREAD

**Issue by spanchang** [w3c/wcag#953]
Refer to examples for SC 2.1.4 under Related Resources at
https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts.html

The examples are very very faint and extremely difficult to discern.
1. Single character key shortcuts affecting speech input – example 1
2. Single character key shortcuts affecting speech input – example 2

Thanks,
Sailesh

**Comment by alastc** [comment]
Hi @spanchang, 

The videos are external resources, not part of the WCAG information. We can ask @KimPatch if she could produce a video at a smaller resolution at a higher quality of recording. However, that is at up to Kim. 

I would note that the examples in the main text cover the same information as the videos.

**Comment by spanchang** [comment]
Hello @alastc  The Understanding doc for the SC describes a situation where the spoken words "Hey Kim" triggers actions within an email app: the Y of "hey" archives the current message and K in "Kim" moves down one conversation.
Explanation for the example of the Understanding doc (Youtube example #2   under Resources on https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts.html)  is: 
"In Twitter, say "James" and the single character key shortcut J moves down one item, the single character key shortcut m opens the message dialog box, and the "es" is printed in the dialog box".
If this  is really happening, is it not the fault of the speech input AT that it parses a word into  separate characters and triggers an action based on a phonetic interpretation? I can't understand how "Hey" can be interpreted as "Y" or "Kim" as "K".
If  a speech input  program (AT) recognizes a shortcut key meant for keyboard action instead of command words, it is something that needs correction / adjustment in the assistive technology. It should not require an action by the content author.
Accidental activation too should not be covered. Surely everyone has experienced calling the wrong person unintentionally or turning on the flash light on an iPhone or perhaps activating a link on a Web page by mistake. 
I believe this SC is making developers and designers  jump through hoops and increasing the cost of testing and ensuring conformance.
Developers build single character key shortcuts to help  certain user groups and now the SC requires them to also incorporate methods by which they can be circumvented. This is not reasonable.
Given that, "Web accessibility depends not only on accessible content but also on accessible Web browsers and other user agents", the onus should be placed on  the makers of AT in this instance.
Best wishes and respectfully,
Sailesh

**Comment by JAWS-test** [comment]
Interestingly, the [European accessibility standard](https://www.etsi.org/deliver/etsi_en/301500_301599/301549/02.01.02_60/en_301549v020102p.pdf) prohibits the exclusive use of shortcuts that require two keys to be pressed simultaneously:

> 5.9 Simultaneous user actions
> Where ICT uses simultaneous user actions for its operation, such ICT shall provide at least one mode of operation that does not require simultaneous user actions to operate the ICT.
> NOTE: Having to use both hands to open the lid of a laptop, **having to press two or more keys** at the same time or having to touch a surface with more than one finger are examples of simultaneous user actions (Page 26)

**Comment by mraccess77** [comment]
In regards to pressing key simultaneously - Most platforms provide a sticky key feature that allows a user to press and release a moodier key.    So if that was supported it seems like it could be used.

**Comment by patrickhlauke** [comment]
I've looked at this old issue again, and have tried to replicate the problems shown in those videos using Windows 11, the built-in Voice Access, and Chrome. Windows VA only sends keystrokes when explicitly told to ("Press d" or similar), or when in an input/text edit. So both the shown scenarios shown in those videos don't happen. (incidentally, back when this SC was being devised, I did push question whether or not that's not a shortcoming/problem of speech recognition software, that it shouldn't just indiscriminately fire off keystroke events unless the user is in a text editing situation...)

> Voice Access error: after saying 'dead' VA shows a note "Did not catch that! Say "What can I say" or if dictating, ensure focus is on a text box.

![screenshot showing google drive in chrome, and windows' voice access bar at the top. after saying 'dead' - as in the video - voice access shows an error](https://github.com/user-attachments/assets/2f704761-8f73-4cb0-9a78-0609e9c772b0)

Will try to test in other configurations, but if this turns out not to be a problem (anymore) I'd suggest simply removing links to those videos from understanding, and possibly clarifying that the issue with speech recognition may not be one anymore (or that it's only with some older/very specific recognition software)...

**Comment by patrickhlauke** [comment]
as an aside, google drive is probably not a good example anymore either, as their keyboard shortcuts now (mostly) use `ctrl` / `shift` / `alt` modifiers... 

![the google drive "Keyboard shortcuts" modal dialog listing all shortcuts, which all use modifier keys](https://github.com/user-attachments/assets/5b1ce6eb-054b-4132-bcff-f26e12d332d6)

**Comment by patrickhlauke** [comment]
after testing this with a few variations (Voice Access on Windows, Voice Control on macOS and iOS, Voice Access on Android, Dragon Home 15 on Windows), I can't replicate the scenario where a page has key event listeners, and these are triggered by accidentally speaking while the speech software is running. All software seems to be context-aware and only accept speech as dictation when it's in a text field. In any other case, unless it's a command, the software does nothing/tells you it didn't understand what you meant.

Testing using my test page https://es-d-78610854020260821-01a00a17-8269-7e91-b0f2-aad0cb1bd835.codepen.dev/

Video showing some of the Windows and Mac testing (didn't bother recording the smartphone testing on iOS/Android, but the results were the same). In all cases, only once focus is in the textarea does freeform speech get turned into text. Also, interestingly, even then it doesn't fire keyboard events, but injects the content directly into the field it seems ... so it doesn't do "fake typing".

not filmed, but: the only way to trigger an arbitrary key press is to explicitly and deliberately say: "Press [name of the key] key". you do see it when I say "Press Tab key" in the video though.

https://github.com/user-attachments/assets/66153d0a-f8d9-4ea3-bb42-5495866c586f

I will open a separate issue/PR to suggest a more fundamental rewrite of the 2.1.4 understanding, pivoting away from the "speech user accidentally triggers commands" scenario which I can't replicate (I am *guessing* based on @KimPatch 's second video that she's using Linux perhaps, and the software there might be less polished and just fire events?) - still mention it, but heavily reduce and caveat it with the fact that in common software these days, that doesn't seem to be a problem - and instead expand on the problem of accidentally triggering commands for keyboard users (and SR users on Windows when there's `role="application"` involved)

If anybody else *is* able to replicate the problem of speech software randomly sending key events, using the above test page, please shout/let me know your exact system/configuration

/cc @alastc

**Comment by patrickhlauke** [comment]
doing a bit of digging around @KimPatch 's archived site and presentations, I *think* she used Dragon Dictate https://en.wikipedia.org/wiki/DragonDictate on Windows 7 (which, reading between the lines, was very patchy when it came to navigation controls etc) and Talon https://talonvoice.com/ (which I just tried to get running on my Win11 system, but had no luck...it simply wasn't doing anything for some reason, despite appearing to sit there in the system tray). It may well be that the issue of speech being turned into "faked" key presses that may trigger single character shortcuts, while present in those early systems, has since been addressed by modern voice/speech control systems.

**Comment by elichad** [comment]
Talon user here! Talon works a bit differently - when you dictate it does trigger key press events. As far as I understand it, Talon does not have an awareness of when you are and aren't in a textarea (which in this comment I'm using as a catch-all for any text input) - dictation mode and command mode are both available at all times and you have to switch between them manually.

Common accidental inputs for me are:
* thinking I'm dictating into a textarea when I'm not, so the dictation triggers keyboard shortcuts instead
* accidentally saying a phrase without prefixing it with the dictation command - in this case Talon tries to interpret the phrase itself as a sequence of commands, which can also cause key presses (sometimes a lot of them, because some commands cause repeats)
* background noise/background speaking being interpreted as commands that trigger key presses

Video attached (as zip file due to size): I activate Talon with Num+ key, `spit` is my Tab command, I dictate short phrases using the command `say <phrase>`. Regardless of where I am on the page it produces key presses. The cursor is going all over the place because I use an eyetracker too, but nothing in the clip depends on the cursor position.

I'm on Windows 11.

[Keyboard listener on document — Mozilla Firefox 2026-08-21 15-55-32.zip](https://github.com/user-attachments/files/31310524/Keyboard.listener.on.document.Mozilla.Firefox.2026-08-21.15-55-32.zip)

**Comment by patrickhlauke** [comment]
Ah perfect, thanks @elichad (and thanks to @eatyourgreens for the intro). so it's still a problem, BUT one that's limited to very specific (non-context-aware) software. This will help shape the rewrite.
